src/map/status.hpp

```diff
@@ -3162,7 +3163,14 @@ struct status_change {
 #ifndef RENEWAL
 	unsigned char sg_counter; //Storm gust counter (previous hits from storm gust)
 #endif
+private:
 	struct status_change_entry *data[SC_MAX];
+public:
+	status_change_entry * getSCE(enum sc_type type) const;
+	status_change_entry * getSCE(uint32 type) const;
+	status_change_entry * createSCE(enum sc_type type);
```

Comment:
I am a little unsure about that one, like 50:50 on it.
I would rather say that this should either have more parameters or let it be created outside.
Otherwise we might end up with uninitialized data?
What do you think?

Reply:
So this PR is just step 1, i can actually change how we do some of our SCE handling if you want, but I wanted this PR to not really change the behavior, add an abstraction